### PR TITLE
US17684-music-migrations

### PR DESCRIPTION
**Pull request by J-Byron**
- Updates song migration to accomodate for crdmusic changes

### CODE REVIEW

File: migrations/20190715104934_update_song.rb [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+      fields = [
+        'description',
+        'details',
+        'chords',
+        'category',
+        'author',
+        'audio_duration',
+        'album',
+        'image',
+        'related_videos',
```
Comment (seancdavis) [inline migrations/20190715104934_update_song.rb]: Recall the note to not remove related videos. https://helloample.slack.com/archives/C0HM9FGKA/p1562692790140900?thread_ts=1562692742.140400&cid=C0HM9FGKA

File: migrations/20190715104934_update_song.rb [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        'tags',
+        'published_at',
+        'unpublished_at',
```
Comment (seancdavis) [inline migrations/20190715104934_update_song.rb]: The publishable fields also should not be removed.

File: migrations/20190715104934_update_song.rb [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        'soundcloud_url',
+        'call_to_action',
+        'featured_subtitle',
+        'featured_label',
+        'duration',
+        'collections'
+      ]
```
Comment (seancdavis) [inline migrations/20190715104934_update_song.rb]: I think it was okay to leave the background image field for now.

File: migrations/20190715104934_update_song.rb [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+      fields.each do |field|
+        content_type.fields.destroy(field)
+      end
+
+      # Create Fields
+      content_type.fields.create(id: 'ccli_number', name: 'CCLI Number', type: 'Symbol', required: true)
+      content_type.fields.create(id: 'written_by', name: 'Written By', type: 'Symbol', required: true)
+      content_type.fields.create(id: 'song_select_url', name: 'Song Select Url', type: 'Symbol')
+      content_type.fields.create(id: 'video', name: 'Video', type: 'Link', link_type: 'Entry', validations: [validation_of_type('video')])
+      content_type.fields.create(id: 'lyric_file', name: 'Lyric File', type: 'Link', link_type: 'Asset')
```
Comment (seancdavis) [inline migrations/20190715104934_update_song.rb]: I would call this `lyrics_file` so we are consistent in our numbering with `chords_file`.

File: migrations/20190715104934_update_song.rb [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+    with_space do |space|
+      
+      # Delete fields
+      fields = [
+        'ccli_number'
+        'written_by'
+        'song_select_url'
+        'video'
+        'lyric_file'
```
Comment (seancdavis) [inline migrations/20190715104934_update_song.rb]: You'll want to change this one to match.

File: migrations/20190715104934_update_song.rb [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+      content_type.fields.create(id: 'author', name: 'Author', type: 'Array', items: items_of_type('Entry', 'author'))	
+      content_type.fields.create(id: 'audio_duration', name: 'Audio Duration', type: 'Text')	
+      content_type.fields.create(id: 'album', name: 'Album', type: 'Link', link_type: 'Entry', validations: [validation_of_type('author')])	
+      content_type.fields.create(id: 'image', name: 'Image', type: 'Link', link_type: 'Asset', required: true)	
+      content_type.fields.create(id: 'featured_subtitle', name: 'Featured subtitle', type: 'Symbol')	
+      content_type.fields.create(id: 'featured_label', name: 'Featured label', type: 'Symbol')	
+      content_type.fields.create(id: 'duration', name: 'duration (seconds)', type: 'Integer')	
+      content_type.fields.create(id: 'collections', name: 'Collections', type: 'Link', link_type: 'Entry', validations: [validation_of_type('collection')])
+      content_type.fields.create(id: 'soundcloud_url', name: 'YouTube URL', type: 'Symbol')	
+      content_type.fields.create(id: 'published_at', name: 'Published At', type: 'Date', required: true)
```
Comment (seancdavis) [inline migrations/20190715104934_update_song.rb]: Remove the ones we don't want to get rid of.

File: migrations/20190715104934_update_song.rb (hunk 1 above)
Comment (J-Byron) [reply migrations/20190715104934_update_song.rb]: Ah, I did not see this. My apologies

**Comment by seancdavis** [comment]
Not all requested changes were made.

File: migrations/20190715112709_update_album.rb [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+      content_type.save
+      content_type.publish
+
+      fields.each do |field|
+        content_type.fields.destroy(field)
+      end
+
+      # Create Fields
+      content_type.fields.create(id: 'behind_the_scenes', name: 'Behind the scenes', type: 'Link', link_type: 'Asset', validations: [require_mime_type(:video)])
```
Comment (seancdavis) [inline migrations/20190715112709_update_album.rb]: This should be just a text (markdown) field. We're going to leave it open-ended for the editors.

File: migrations/20190715112709_update_album.rb [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+      content_type.fields.create(id: 'songs', name: 'Songs', type: 'Array', items: items_of_type('Entry', 'song'))
+
```
Comment (seancdavis) [inline migrations/20190715112709_update_album.rb]: We also now want to add featured videos as a field to albums.

**Pull request merged**